mobile/src/main/java/org/openhab/habdroid/ui/OpenHABPreferencesActivity.java

```diff
@@ -224,17 +243,86 @@ public boolean onPreferenceClick(Preference preference) {
-                final PreferenceScreen ps = getPreferenceScreen();
-                ps.removePreference(ps.findPreference(Constants.PREFERENCE_FULLSCREEN));
+            ringtonePreference.setOnPreferenceChangeListener(new OnPreferenceChangeListener() {
+                @Override
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    updateRingtonePreferenceSummary(preference, newValue);
+                    return true;
+                }
+            });
+
+            final PreferenceScreen ps = getPreferenceScreen();
+
+            // Fullscreen is not supoorted in Android < 4.4
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                Preference fullscreenPreference = ps.findPreference(Constants.PREFERENCE_FULLSCREEN);
+                fullscreenPreference.setEnabled(false);
+                fullscreenPreference.setSummary(R.string.settings_disabled_kitkat);
+            }
+
+            removeUnsupportedPrefs(ps);
+
+            if (mOpenhabVersion == 1) {
+                Preference iconFormatPreference =
+                        ps.findPreference(Constants.PREFERENCE_ICON_FORMAT);
+                iconFormatPreference.setEnabled(false);
+                iconFormatPreference.setSummary(R.string.settings_disabled_openhab_1);
+                Preference chartScalingPreference =
+                        ps.findPreference(Constants.PREFERENCE_CHART_SCALING);
+                chartScalingPreference.setEnabled(false);
+                chartScalingPreference.setSummary(R.string.settings_disabled_openhab_1);
             }
         }
 
         private void onNoDefaultSitemap(Preference pref) {
             pref.setEnabled(false);
             pref.setSummary(R.string.settings_no_default_sitemap);
         }
+
+        private void updateRingtonePreferenceSummary(Preference pref, Object newValue) {
+            String value = (String) newValue;
+            if (TextUtils.isEmpty(value)) {
+                pref.setSummary(R.string.settings_ringtone_none);
+            } else {
+                Ringtone ringtone = RingtoneManager.getRingtone(getActivity(), Uri.parse(value));
+                if (ringtone != null) {
+                    pref.setSummary(ringtone.getTitle(getActivity()));
+                }
+            }
+        }
+
+        public void updateConnectionSummary(boolean isRemote) {
+            String subscreenPrefValue = isRemote ? Constants.SUBSCREEN_REMOTE_CONNECTION
+                    : Constants.SUBSCREEN_LOCAL_CONNECTION;
+            String urlPrefValue = isRemote ? Constants.PREFERENCE_REMOTE_URL
+                    : Constants.PREFERENCE_LOCAL_URL;
+            String userPrefValue = isRemote ? Constants.PREFERENCE_REMOTE_USERNAME
+                    : Constants.PREFERENCE_LOCAL_USERNAME;
+            String passwordPrefValue = isRemote ? Constants.PREFERENCE_REMOTE_PASSWORD
+                    : Constants.PREFERENCE_LOCAL_PASSWORD;
```

Comment:
Could pass the 4 key names as parameters here ... that way it would be a bit more clear at the caller side what's going on.

Reply:
Done